### ISSUE TITLE
VSCode automatically removes content that has syntax error

### ISSUE THREAD

**Issue by ymc9** [zenstackhq/zenstack#325]
In VSCode editor with a .zmodel file, past the following:

```
datasource db {
    provider = 'sqlite'
    url = 'dev.db'
}

model User {
    id Int @id @default(autoincrement())
    username String
    sent Chat[] @relation('sent')
    received Chat[] @relation('received')
}

plugg abc {
    
}

model Chat {
    id Int @id @default(autoincrement())
    subject String
    fromUser User @relation(name: 'sent', fields: [fromUserId], references: [id])
    fromUserId Int
    toUser User @relation(name: 'received', fields: [toUserId], references: [id])
    toUserId Int
}
```

When saved, the "plugg ABC { ... }" part disappears.

**Comment by ymc9** [comment]
@jiashengguo do you think this can be related to auto-formatting?

**Comment by jiashengguo** [comment]
Yes. It seems that langium has got a fix for it, but targeted for the next release 1.2.0:
 #https://github.com/langium/langium/issues/912
 
The idea is simply not to format if there is any syntax error.  I think I can make the same fix from our side first.

**Comment by ymc9** [comment]
fixed by #407
